development-guidelines/incident_response.md

```diff
@@ -1,48 +1,58 @@
-  - **Decentralized**. Using decentralized components can sometimes restrict rescue measures.
-- **Evaluate what events are needed**. A missed event in a critical location might result in unnoticed incidents.
+  - **Decentralized**. Using decentralized components can sometimes limit what rescue measures are possible and may require a higher amount of coordination.
+- **Begin to identify important system invariants**. This helps to determine what you will need to monitor and what events may be necessary to do so effectively.
+- **Evaluate what additional events are needed**. A missed event in a critical location might result in unnoticed incidents.
 - **Evaluate what components must be on-chain and off-chain**. On-chain components are generally more at risk, but off-chain components push the risks to the off-chain owner.
-- **Use an access control that allows fine-grained access**. Avoid setting all access controls to be available to an EOA. Opt for multisig wallets/MPC, and segregate access (e.g., the key responsible for setting fees shouldn't have access to the upgradeability feature).
+- **Use fine-grained access controls**. Avoid setting all access controls to be available to an EOA. Opt for multisig wallets/MPC, and avoid delegating several roles to one address (e.g., the key responsible for setting fees shouldn't have access to the upgradeability feature).
 
 ## Documentation
 
-- **Document how to interpret abnormal events emission**. Only emitting events isn't sufficient; proper documentation is crucial, and users should be empowered to decode them.
-- **Document how to access wallets**. Clearly outline how to access wallets. Both the location and access procedures for every wallet should be clear and straightforward.
-- **Document the deployment and upgrade process**. Deployment and upgrade processes are risky and must be thoroughly documented. This should include how to test the deployment/upgrade (ex: using fork testing) and how to validate it (ex: using a post-deployment script).
-- **Document how to contact the users and external dependencies**. Define guidelines regarding which stakeholders to contact, including the timing and mode of communication in case of incidents.
+- **Assemble a runbook of common actions you may need to perform**. It's not possible or practical to exhaustively detail how you'll respond to every type of incident. But you _can_ start to document procedures for some of the more important ones as well as actions that might be common across multiple scenarios (e.g., pausing, rotating owner keys, upgrading an implementation). This can also include scripts or snippets of code to facilitate performing these actions in a reproducible manner.
+- **Document how to interpret abnormal events emission**. Only emitting events isn't sufficient; proper documentation is crucial, and users should be empowered to identify and decode them.
```

Comment:
This point could be more clear:
- The point seems to be focused on documenting how to interpret "abnormal" events. 
- The text that follows seems more generally focused on documenting events. 

Consider removing the word "abnormal" from the bolded text. 

Also consider starting the non-bolded text with "Emitting events isn't sufficient..." (drop the "Only" as it sounds redundant. )